bokehjs/examples/stocks/stocks.ts

```diff
@@ -0,0 +1,73 @@
+console.log(`Bokeh ${Bokeh.version}`);
+Bokeh.set_log_level("debug");
+
+
+function make_plot(title:string, source:Bokeh.ColumnDataSource) {
+    
+    // Create plot
+    var xr = new Bokeh.DataRange1d({});
+    var yr = new Bokeh.DataRange1d({});
+    var plot = new Bokeh.Plot({        
+        x_range: xr,
+        y_range: yr,         
+        title: title,
+        plot_width: 400,
+        plot_height: 400,
+        border_fill_color: "#eeeeff",
+    });
+    
+    // Add axis and grid
+    const xaxis = new Bokeh.DatetimeAxis({axis_line_color: null, axis_label: 'time'});
+    const yaxis = new Bokeh.LinearAxis({axis_line_color: null, axis_label: 'price'});
+    plot.add_layout(xaxis, "below");
+    plot.add_layout(yaxis, "left");
+    plot.add_layout(new Bokeh.Grid({ticker: xaxis.ticker, dimension: 0}));
+    plot.add_layout(new Bokeh.Grid({ticker: yaxis.ticker, dimension: 1}));
+    
+    // Add a line for each entry in the source
+    var colors = ['#aa0000','#00aa00', '#0000aa', '#aaaa00', '#aa00aa', '#00aaaa'];
+    var i = -1;
+    for (let key in source.data) {
+        if (key != 't') {
+            i += 1;
+            let line = new Bokeh.Line({x:{field:'t'}, y:{field:key}, legend: key, 
+                                       line_color: colors[i%6], line_width: 2});
+            plot.add_glyph(line, source);
+        }
+    }
+    
+    // Add tools
+    plot.add_tools(new Bokeh.PanTool(), new Bokeh.WheelZoomTool());
+    
+    // Tweaks to work around bootstrap problems 
+    xr.renderers = plot.renderers;
+    yr.renderers = plot.renderers;
```

Comment:
So this is why `tap` example fails. Initialization of range (and tools) need some work to make it robust.